### PR TITLE
[bitnami/kafka] Improved compatibility with Kubernetes 1.21+ and Openshift

### PR DESCRIPTION
**Pull request by hierynomus**
**Description of the change**
There are 2 changes to the kafka chart:
- Ensure that the chart is Kubernetes 1.21+ compatible by making sure that the PodDisruptionBudget uses the `policy/v1` apiVersion if it is available.
- Can set a special `auto` value to the `volumePermissions.securityContext.runAsUser` to make it OpenShift compatible as that does not like the `0` (root) user.

**Benefits**
Compatibility with newer Kubernetes versions and OpenShift deployments

**Possible drawbacks**
Fully backwards compatible so no drawbacks expected.>

**Applicable issues**
N/A

**Additional information**
N/A

**Checklist** 

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])

### CODE REVIEW

File: bitnami/kafka/values.yaml [hunk 1]
```diff
@@ -902,7 +902,7 @@ volumePermissions:
   ##
   enabled: false
   ## The security context for the volumePermissions init container
-  ## @param volumePermissions.securityContext.runAsUser User ID for the container
+  ## @param volumePermissions.securityContext.runAsUser User ID for the container. Can be set to "auto". "auto" is especially useful for OpenShift which has scc with dynamic userids (and 0 is not allowed).
```
Comment (juan131) [inline bitnami/kafka/values.yaml:905]: Split the comment:
```suggestion
  ## @param volumePermissions.securityContext.runAsUser User ID for the container.
  ## Can be set to "auto". "auto" is especially useful for OpenShift which has scc with dynamic user ids (and 0 is not allowed).
```

File: bitnami/kafka/templates/poddisruptionbudget.yaml [hunk 2]
```diff
@@ -1,6 +1,10 @@
 {{- $replicaCount := int .Values.replicaCount }}
 {{- if and .Values.pdb.create (gt $replicaCount 1) }}
+{{- if .Capabilities.APIVersions.Has "policy/v1/PodDisruptionBudget" }}
+apiVersion: policy/v1
+{{- else }}
 apiVersion: policy/v1beta1
+{{- end }}
```
Comment (juan131) [inline bitnami/kafka/templates/poddisruptionbudget.yaml:7]: We have a [named templated](https://github.com/bitnami/charts/blob/master/bitnami/common/templates/_capabilities.tpl#L21) available in the `common` chart that you can use for this:
```suggestion
apiVersion: {{ include "common.capabilities.policy.apiVersion" . }}
```

**Review by juan131 — CHANGES_REQUESTED** [review]
Thanks so much for this great PR @hierynomus !! Please check my comments when you have a chance.

**Comment by hierynomus** [comment]
@juan131 Thanks for the review. I've committed your suggested changes.

**Review by juan131 — APPROVED** [review]
LGTM

**Pull request merged**